### DISCUSSION TITLE
Sync fork without overwriting custom key mapping

### DISCUSSION THREAD

**Discussion by fuchodeveloper** [KinesisCorporation/Adv360-Pro-ZMK#413] [Q&A]
Hi, I have a couple of custom key mappings which match my workflow but after syncing my forked branch with the upstream from Kinesis main branch, there are usually conflicts and I have to resolve them using the ADV PRO GUI by remapping the keys.
Is there a more better way of resolving this issue by persisting my custom keymaps after each sync?

**Comment by ReFil** [comment]
What custom keymapping are you using? We try to do updates that do not cause merge conflicts

**Reply by fuchodeveloper** [reply]
Hey @ReFil, I totally missed the reply here. I have custom mapping for the Chromium-based browsers, Safari, media keys etc. Because these keys already have default mappings, whenever I update my branch with the Kinesis origin and install the latest changes, I have to repeat the steps of adding my custom mapping

**Reply by ReFil** [reply]
If you're following the recommended branch update procedure through GitHub on the browser it shouldn't have conflicts. We haven't made changes to the keymap file since implementing the versioning code

**Reply by fuchodeveloper** [reply]
This is an example message that shows up when syncing with the Kinesis origin: 
```
This branch is out-of-date
Update branch to merge the latest changes from the upstream repository into this branch.

Discard 3 commits to make this branch match the upstream repository. 3 commits will be removed from this branch.
```
I assume you're referring to https://github.com/KinesisCorporation/Adv360-Pro-ZMK/blob/V3.0/UPGRADE.md

Although I'm already on V3, I'll give the guide a shot

**Reply by ReFil** [reply]
If you click "sync fork" then "update branch" there shouldn't be conflicts. We've deliberately tried to minimise the potential conflicts. Upgrade.md is only for going from v2.0 to v3.0

**Reply by ReFil** [reply]
https://github.com/fuchodeveloper/Adv360-Pro-ZMK/compare/V3.0...KinesisCorporation%3AAdv360-Pro-ZMK%3AV3.0 I ran a quick compare of your branch to upstream and there are no conflicts, you should be able to just merge it in
